src/runtime/methodbinder.cs

```diff
@@ -455,6 +467,18 @@ internal Binding Bind(IntPtr inst, IntPtr args, IntPtr kw)
             }
             return null;
         }
+        /// <summary>
+        /// Returns true if src type (int -type) is promotable to a higher resolution type (float|double)
+        /// </summary>
+        /// <param name="src"></param>
+        /// <param name="dst"></param>
+        /// <returns></returns>
+        private bool SimplePromotableType(Type src, Type dst)
+        {
+            return (((src == typeof(int) || src == typeof(short)) && (dst == typeof(double) || dst == typeof(float))))
+                || ((src == typeof(long)) && (dst == typeof(double)))
```

Comment:
My point was that *dis*allowing the conversion `long -> float` doesn't make sense as the conversion `long -> double` is also lossy (64 vs 53). So either we allow all of those or none.